### PR TITLE
added role based access, User entity, JWT-based authentication,  fixe…

### PR DESCRIPTION
**Pull request by DmytroFedoryshyn**
…d book sorting

### CODE REVIEW

**Comment by DmytroFedoryshyn** [comment]
For some reason, CI checks fail. When I run mvn clean package on my machine, it works perfectly

**Comment by aaananas** [comment]
Did you push all the changes? pom.xml etc

File: src/main/java/bookstore/dto/UserLoginRequestDto.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.Size;
+import lombok.Data;
+
+@Data
+public class UserLoginRequestDto {
+    @Email
+    @NotBlank
+    @Size(min = 8, max = 50)
```
Comment (Rommelua) [inline src/main/java/bookstore/dto/UserLoginRequestDto.java]: there are all checks inside @Email
```suggestion
```

File: src/main/java/bookstore/dto/UserRegistrationRequest.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.Size;
+import lombok.Data;
+
+@Data
+@PasswordsMatch
+public class UserRegistrationRequest {
+    @Email
+    @Size(min = 4, max = 50)
```
Comment (Rommelua) [inline src/main/java/bookstore/dto/UserRegistrationRequest.java]:
```suggestion
```

File: src/main/java/bookstore/dto/UserRegistrationRequest.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    private String email;
+    @NotBlank
+    @Size(min = 6, max = 100)
+    private String password;
+    @NotBlank
+    @Size(min = 6, max = 100)
```
Comment (Rommelua) [inline src/main/java/bookstore/dto/UserRegistrationRequest.java]:
```suggestion
```

File: src/main/java/bookstore/model/User.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    }
+
+    @Override
+    public boolean isCredentialsNonExpired() {
+        return true;
+    }
+
+    @Override
+    public boolean isEnabled() {
+        return true;
```
Comment (Rommelua) [inline src/main/java/bookstore/model/User.java]:
```suggestion
        return !isDeleted;
```

File: src/main/java/bookstore/repository/user/UserRepository.java [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+import bookstore.model.User;
+import java.util.Optional;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface UserRepository extends JpaRepository<User, Long> {
+    @Query("FROM User u LEFT JOIN FETCH u.roles WHERE u.email = :email AND u.isDeleted = false")
```
Comment (Rommelua) [inline src/main/java/bookstore/repository/user/UserRepository.java]:
```suggestion
    @EntityGraph(attributePaths = "roles")
```

**Review by Rommelua — CHANGES_REQUESTED** [review]

**Pull request merged**